Release blogpost for v11.1 (#817)
rescript-association/rescript-lang.org: _blogposts/2024-02-01-release-11-1.mdx

---
author: rescript-team
date: "2024-02-29"
previewImg: /static/blog/compiler_release_11_0.jpg
title: ReScript 11.1
badge: release
description: |
  Unleashing ReScript from React
---

> A couple of weeks ago, the ReScript team released ReScript 11.0, which laid ground work for a lot of possible improvements to make it easier to interact with the JavaScript ecosystem. It also came with a lot of breaking changes, unfortunately.

Reviewer: Not sure if we should emphasize those breaking changes that much. :-)

Reviewer 2: Agreed. I don't think we need to say anything about the breaking changes at all, enough that they were mentioned in the v11 post.

> A couple of weeks ago, the ReScript team released ReScript 11.0, which laid ground work for a lot of possible improvements to make it easier to interact with the JavaScript ecosystem. It also came with a lot of breaking changes, unfortunately.

Author: I removed the mention of breaking changes instead.

## Import attributes

Import attributes is a JS feature that is [currently in standardization](https://github.com/tc39/proposal-import-attributes), but is already implemented by many JS tools. Now, ReScript supports it too, as long as the compiler is configured to output ES6.

<CodeTab labels={["ReScript", "JS Output"]}>

```res
@module({from: "./myJson.json", with: {type_: "json", \"some-identifier": "yep"}})
external myJson: Js.Json.t = "default"
Console.log(myJson)

@module({from: "./myCss.css", with: {type_: "css", \"some-identifier": "yep"}})
external buttonCss: string = "button"
Console.log(buttonCss)
```
```js
import * as MyCssCss from "./myCss.css" with {"type": "css", "some-identifier": "yep"};
import MyJsonJson from "./myJson.json" with {"type": "json", "some-identifier": "yep"};

var myJson = MyJsonJson;
console.log(myJson);

var buttonCss = MyCssCss.button;
console.log(buttonCss);
```
</CodeTab>

## Array spread syntax

The spread syntax, which was already supported for records and lists for a long time, now also supports arrays!

```rescript
let animals = ["🐶", "🐱", "🐷"]
let moreAnimals = [...animals, "🐔", "🐴", "🐮"]
````

## Hyphens in JSX tag names

> We lifted restrictions on JSX tag names. This means you no longer need to use the [illegal identifier syntax](/docs/manual/latest/use-illegal-identifier-names) to use tag names that contain hyphens:

Reviewer: ```suggestion
We lifted restrictions on JSX tag names. This means you no longer need to escape tag names that contain hyphens:
```

Previously:

```rescript
let x = <\"custom-tag" />
let y = <Foo.\"custom-tag" />
```

Now:

```rescript
let x = <custom-tag />

> let y= <Foo.custom-tag />

Reviewer: ```suggestion
let y = <Foo.custom-tag />
```

```

## Acknowledgements

Once again we want to thank everyone from the community who volunteered their precious time to support this project with contributions of any kind, from documentation, to PRs, to discussions in the forum. But especially we want to thank the following people, who helped landing this release:

> [cknitt](https://github.com/cknitt), [@cometkim](https://github.com/cometkim), [@cristianoc](https://github.com/cristianoc), [@diogomqbm](https://github.com/diogomqbm), [@kevinbarabash](https://github.com/kevinbarabash), [@shulhi](https://github.com/shulhi), [@tsnobip](https://github.com/tsnobip), [@zth](https://github.com/zth).

Reviewer: ```suggestion
[@cknitt](https://github.com/cknitt), [@cometkim](https://github.com/cometkim), [@cristianoc](https://github.com/cristianoc), [@diogomqbm](https://github.com/diogomqbm), [@kevinbarabash](https://github.com/kevinbarabash), [@shulhi](https://github.com/shulhi), [@tsnobip](https://github.com/tsnobip), [@zth](https://github.com/zth).
```
